validator/validator/metrics.py

```diff
@@ -1,65 +1,71 @@
+from dataclasses import dataclass
+
 import bittensor as bt
 
-from neuron import CheckpointBenchmark
+from neuron import CheckpointBenchmark, CURRENT_CONTEST
+from .validator import Uid
+
+
+@dataclass
+class MetricData:
+    baseline_average: float
+    model_average: float
+    similarity_average: float
+    size: int
+    vram_used: float
+    watts_used: float
+
+    def calculate_score(self) -> float:
+        return max(
+            0.0,
+            (self.baseline_average or CURRENT_CONTEST.baseline_average) - self.model_average
+        ) * self.similarity_average
+
+    def __repr__(self):
+        return f"MetricData(baseline_average={self.baseline_average}, model_average={self.model_average}, similarity_average={self.similarity_average}, size={self.size}, vram_used={self.vram_used}, watts_used={self.watts_used})"
+
+
+def new_metric_data() -> MetricData:
```

Comment:
Just set default values in the data class(`name: type = default`), and then use the default constructor.